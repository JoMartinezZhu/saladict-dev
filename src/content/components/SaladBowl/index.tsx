import React from 'react'

export interface SaladBowlProps {
  readonly mouseOnBowl: (flag: boolean) => any
  readonly isAnimation: boolean
  readonly bowlHover: boolean
}

export default class SaladBowl extends React.PureComponent<SaladBowlProps> {
  mouseOnBowlTimeout: any

  handleMouseEnter = () => {
    if (this.props.bowlHover) {
      this.mouseOnBowlTimeout = setTimeout(() => {
        this.props.mouseOnBowl(true)
      }, 500)
    }
  }

  handleMouseLeave = () => {
    clearTimeout(this.mouseOnBowlTimeout)
    this.props.mouseOnBowl(false)
  }

  handleClick = () => {
    this.props.mouseOnBowl(true)
  }

  render () {
    const { isAnimation, bowlHover } = this.props
    return (
      <div className={(
        'saladict-SaladBowl' +
        (isAnimation ? ' isAnimate' : '') +
        (bowlHover ? ' bowlHover' : '')
      )}
        role='img'
        key='saladict-SaladBowl'
        onMouseEnter={this.handleMouseEnter}
        onMouseLeave={this.handleMouseLeave}
        onClick={this.handleClick}
      >
        <div className='saladict-SaladBowl_Leaf' />
        <div className='saladict-SaladBowl_Orange' />
        <div className='saladict-SaladBowl_Tomato' />
        <div className='saladict-SaladBowl_Bowl' />
      </div>
    )
  }
}
