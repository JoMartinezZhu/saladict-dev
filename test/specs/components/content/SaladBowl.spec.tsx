import React from 'react'
import { shallow } from 'enzyme'
import noop from 'lodash/noop'

import SaladBowl, { SaladBowlProps } from '@/content/components/SaladBowl'

describe('Component/content/SaladBowl', () => {
  it('should render correctly', () => {
    const props: SaladBowlProps = {
      isAnimation: true,
      mouseOnBowl: noop,
    }
    expect(shallow(<SaladBowl {...props}/>)).toMatchSnapshot()
  })
})
