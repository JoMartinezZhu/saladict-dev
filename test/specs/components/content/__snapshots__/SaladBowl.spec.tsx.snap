// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/content/SaladBowl should render correctly 1`] = `
<div
  className="saladict-SaladBowl isAnimate"
  key="saladict-SaladBowl"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="img"
>
  <div
    className="saladict-SaladBowl_Leaf"
  />
  <div
    className="saladict-SaladBowl_Orange"
  />
  <div
    className="saladict-SaladBowl_Tomato"
  />
  <div
    className="saladict-SaladBowl_Bowl"
  />
</div>
`;
